src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,236 @@
+use std::iter;
+use std::io::{Result, Write};
+
+use analysis::namespaces::MAIN;
+use env::Env;
+use file_saver::save_to_file;
+use library::{self, Library, ToStripperType};
+use library::Type as LType;
+
+use stripper_interface::{FILE, /*FILE_COMMENT, */MOD_COMMENT, TypeStruct};
+
+const TAB_SIZE : usize = 4;
+const DEPRECATED : &'static str = "/// ====>> DEPRECATED <<====";
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(mut w: &mut Write, lib: &Library) -> Result<()> {
+    let namespace = lib.namespace(MAIN);
+
+    /*if let Some(ref namespace_doc) = namespace.doc {
+        for line in namespace_doc.split("\n") {
+            output.push_str(&format!("{}{}\n", FILE_COMMENT, line));
+        }
+    }
+    try!(create_sub_docs(&mut output, &namespace.constants, None));*/
+    for ty in namespace.types.iter().filter_map(|t| t.as_ref()) {
+        try!(handle_type(&mut w, &ty, None, lib));
+    }
+    Ok(())
+}
+
+fn handle_type(w: &mut Write, ty: &library::Type, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    match *ty {
+        LType::Alias(ref a) => {
+            try!(writeln!(w, "{}src/auto/{}.rs", FILE, a.name.to_lowercase()));
+            create_sub_doc(w, a, parent)
+        },
+        LType::Enumeration(ref e) => create_enum_doc(w, &e, parent),
+        LType::Function(ref f) => create_fn_doc(w, &f, parent),
+        LType::Interface(ref i) => create_trait_doc(w, &i, parent),
+        LType::Class(ref c) => create_class_doc(w, &c, parent, lib),
+        _ => Ok(()),
+    }
+}
+
+fn create_class_doc(w: &mut Write, class: &library::Class, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    let tabs : String = iter::repeat(" ").take(compute_ident(&parent) * TAB_SIZE).collect::<Vec<&str>>().concat();
+    let mut ty = class.convert();
+    ty.parent = parent;
+
+    try!(writeln!(w, "{}src/auto/{}.rs", FILE, class.name.to_lowercase()));
+    let found = if let Some(ref class_doc) = class.doc {
+        try!(writeln!(w, "{}{}\n{}",
+                      MOD_COMMENT,
+                      ty,
+                      format!("{}/// {}",
+                              tabs,
+                              class_doc.split("\n").collect::<Vec<&str>>().join(&format!("\n{}/// ", tabs)))));
```

Comment:
https://github.com/gtk-rs/gir/blob/master/src/file_saver.rs#L18

This reminds me: notice the `Untabber` filter. You can avoid messing with tab size and `"    "` because a '\t' will get expanded properly.